server/src/test/java/org/elasticsearch/index/mapper/WholeNumberFieldMapperTests.java

```diff
@@ -22,4 +24,44 @@ protected void testDecimalCoerce() throws IOException {
         assertEquals(7, pointField.numericValue().doubleValue(), 0d);
     }
 
+    @Override
+    public void testDimension() throws IOException {
+        MapperService mapperService = createMapperService(fieldMapping(b -> {
+            minimalMapping(b);
+            b.field("dimension", true);
+        }));
+
+        MappedFieldType fieldType = mapperService.fieldType("field");
+        assertThat(fieldType, instanceOf(NumberFieldMapper.NumberFieldType.class));
+        NumberFieldMapper.NumberFieldType ft = (NumberFieldMapper.NumberFieldType) fieldType;
+        assertTrue(ft.isDimension());
+    }
+
+    @Override
+    protected void registerParameters(ParameterChecker checker) throws IOException {
+        super.registerParameters(checker);
+
+        // dimension cannot be updated
+        checker.registerConflictCheck("dimension", b -> b.field("dimension", true));
+        checker.registerConflictCheck("dimension", b -> b.field("dimension", false));
+        checker.registerConflictCheck("dimension",
+            fieldMapping(b -> {
+                minimalMapping(b);
+                b.field("dimension", false);
+            }),
+            fieldMapping(b -> {
+                minimalMapping(b);
+                b.field("dimension", true);
+            }));
+        checker.registerConflictCheck("dimension",
+            fieldMapping(b -> {
+                minimalMapping(b);
+                b.field("dimension", true);
+            }),
+            fieldMapping(b -> {
+                minimalMapping(b);
+                b.field("dimension", false);
+            }));
```

Comment:
I wonder if we could stick this in a method on `MapperTestCase` or something. That way all of the fields with the `dimension` parameter could add these with a single line. That way its a little more obvious that we expect them to work the same way.